numba/types/abstract.py

```diff
@@ -209,9 +249,213 @@ class Hashable(Type):
     """
     Base class for hashable types.
     """
+    @classmethod
+    def interface_check(cls, instance):
```

Comment:
So, instead of adding all the interface checking machinery, is it doable to simply have separate hashing declaration and implementation for jitclass instances?